src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -196,6 +197,31 @@ param(
     default { throw "7-Zip signalled an unknown error (code $exitCode)" }
   }
 
-  $env:ChocolateyPackageInstallLocation = $destination
-  return $destination
+  if ((test-path variable:url) -AND (@($url).count -gt 0))

```

Comment:
I had submitted a question about this at the time - but I believe $url was the only reliable way (variable was always available) to find the original zip name.  I think packagename is different than the original zip name by this point in processing and another internal variable was not created in every situation.

As expressed in further on comments - I was keeping the scope of what was looked for and unzipped as tight as possible - so hence the need to have the original archive name so as to only look for matching named sub-archives.